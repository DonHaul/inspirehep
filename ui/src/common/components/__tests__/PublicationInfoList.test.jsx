import { render } from '@testing-library/react';
import { fromJS } from 'immutable';

import PublicationInfoList from '../PublicationInfoList';

describe('PublicationInfoList', () => {
  it('renders with publicationInfo', () => {
    const publicationInfo = fromJS([
      {
        journal_title: 'Test Journal',
      },
    ]);
    const { getByText } = render(
      <PublicationInfoList publicationInfo={publicationInfo} />
    );
    expect(getByText(/Test Journal/i)).toBeInTheDocument();
    expect(getByText(/Published in:/i)).toBeInTheDocument();
  });

  it('renders without label if labeled false', () => {
    const publicationInfo = fromJS([
      {
        journal_title: 'Test Journal',
      },
    ]);
    const { queryByText } = render(
      <PublicationInfoList labeled={false} publicationInfo={publicationInfo} />
    );
    expect(queryByText(/Published in:/i)).toBeNull();
  });

  // TODO: wrapperClassName is not being used in the component fix test
  it('renders with wrapperClassName', () => {
    const publicationInfo = fromJS([
      {
        journal_title: 'Test Journal',
      },
    ]);
    const { asFragment } = render(
      <PublicationInfoList
        wrapperClassName="test"
        publicationInfo={publicationInfo}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
