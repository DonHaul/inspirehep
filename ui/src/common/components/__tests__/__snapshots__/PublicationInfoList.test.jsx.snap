// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationInfoList renders with wrapperClassName 1`] = `
<DocumentFragment>
  <div
    class="__InlineList__"
    data-testid="inline-data-list"
  >
    <span
      class=""
    >
      Published in: 
    </span>
    <ul>
      <li>
        <span>
          <span>
            <i>
              Test Journal
            </i>
          </span>
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
